Sources/Widgets/Widgets3D/ResliceCursorWidget/index.js

```diff
@@ -0,0 +1,562 @@
+    // reslice cursor is. This allows us to capture the whole image even
+    // if we resliced in awkward places.
+    const offset = [];
+    for (let i = 0; i < 3; i++) {
+      offset[i] = -Math.abs(center[i] - imageCenter[i]);
+      offset[i] *= 2; // give us room
+    }
+    // Add a small offset to force the recomputation of the 3 plane points (origin
+    // point1 and point2) after setting the normal in function updateReslicePlane
+    // Else, the three points won't be in adequation with the set normal
+    if (offset[0] === 0 && offset[1] === 0 && offset[2] === 0) {
+      offset[0] += 0.0000001;
+      offset[1] += 0.0000001;
+      offset[2] += 0.0000001;
+    }
+
+    // Now set the size of the plane based on the location of the cursor so as to
+    // at least completely cover the viewed region
+    const planeSource = vtkPlaneSource.newInstance();
+
+    if (viewType === ViewTypes.CORONAL) {
+      planeSource.setOrigin(
+        bounds[0] + offset[0],
+        center[1],
+        bounds[4] + offset[2]
+      );
+      planeSource.setPoint1(
+        bounds[1] - offset[0],
+        center[1],
+        bounds[4] + offset[2]
+      );
+      planeSource.setPoint2(
+        bounds[0] + offset[0],
+        center[1],
+        bounds[5] - offset[2]
+      );
+    } else if (viewType === ViewTypes.AXIAL) {
+      planeSource.setOrigin(
+        bounds[0] + offset[0],
+        bounds[2] + offset[1],
+        center[2]
+      );
+      planeSource.setPoint1(
+        bounds[1] - offset[0],
+        bounds[2] + offset[1],
+        center[2]
+      );
+      planeSource.setPoint2(
+        bounds[0] + offset[0],
+        bounds[3] - offset[1],
+        center[2]
+      );
+    } else if (viewType === ViewTypes.SAGITTAL) {
+      planeSource.setOrigin(
+        center[0],
+        bounds[2] + offset[1],
+        bounds[4] + offset[2]
+      );
+      planeSource.setPoint1(
+        center[0],
+        bounds[3] - offset[1],
+        bounds[4] + offset[2]
+      );
+      planeSource.setPoint2(
+        center[0],
+        bounds[2] + offset[1],
+        bounds[5] - offset[2]
+      );
+    }
+    return planeSource;
+  }
+
+  function updateCamera(renderer, normal) {
+    // When the reslice plane is changed, update the camera to look at the
+    // normal to the reslice plane always.
+
+    const focalPoint = renderer.getActiveCamera().getFocalPoint();
+    const position = renderer.getActiveCamera().getPosition();
+
+    const distance = Math.sqrt(
+      vtkMath.distance2BetweenPoints(position, focalPoint)
+    );
+
+    const estimatedCameraPosition = [
+      focalPoint[0] + distance * normal[0],
+      focalPoint[1] + distance * normal[1],
+      focalPoint[2] + distance * normal[2],
+    ];
+
+    // intersect with the plane to get updated focal point
+    const intersection = vtkPlane.intersectWithLine(
+      focalPoint,
+      estimatedCameraPosition,
+      model.widgetState.getCenter(),
+      normal
```

Comment:
odd that you have normal as parameter but not origin. You should be consistent

Reply:
Not sure to understand. Did you mean that I should replace `model.widgetState.getCenter()` by a constant named origin ?

Reply:
Me neither, I do not remember what I had in mind